drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -334,29 +341,61 @@ local function mired_to_kelvin(value, minOrMax)
   end
 end
 
+local function is_supported_combination_button_switch_device_type(device, endpoint_id)
+  for _, ep in ipairs(device.endpoints) do
+    if ep.endpoint_id == endpoint_id then
+      for _, dt in ipairs(ep.device_types) do
+        if dt.device_type_id == DIMMABLE_LIGHT_DEVICE_TYPE_ID then
+          return true
+        end
+      end
+    end
+  end
+  return false
+end
+
+local function get_first_non_zero_endpoint(endpoints)
+  for _,ep in ipairs(endpoints) do
+    if ep ~= 0 then -- 0 is the matter RootNode endpoint
+      return ep
+    end
+  end
+  return nil
+end
+
 --- find_default_endpoint helper function to handle situations where
 --- device does not have endpoint ids in sequential order from 1
 --- In this case the function returns the lowest endpoint value that isn't 0
 --- and supports the OnOff or Switch cluster. This is done to bypass the
 --- BRIDGED_NODE_DEVICE_TYPE on bridged devices.
-local function find_default_endpoint(device, component)
+local function find_default_endpoint(device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH})
-  local all_eps = {}
+  table.sort(switch_eps)
+  table.sort(button_eps)
 
-  for _,ep in ipairs(switch_eps) do
-    table.insert(all_eps, ep)
+  -- Return the first switch endpoint as the default endpoint if no button endpoints are available
+  if #button_eps == 0 and #switch_eps > 0 then
+    return get_first_non_zero_endpoint(switch_eps)
   end
-  for _,ep in ipairs(button_eps) do
-    table.insert(all_eps, ep)
+
+  -- Return the first button endpoint as the default endpoint if no switch endpoints are available
+  if #switch_eps == 0 and #button_eps > 0 then
+    return get_first_non_zero_endpoint(button_eps)
   end
-  table.sort(all_eps)
 
-  for _, ep in ipairs(all_eps) do
-    if ep ~= 0 then --0 is the matter RootNode endpoint
-      return ep
+  -- If both switch and button endpoints are present, check the device type on the main switch
+  -- endpoint. If it is not a supported device type, return the first button endpoint as the
+  -- default endpoint.
+  if #switch_eps > 0 and #button_eps > 0 then
+    local main_endpoint = get_first_non_zero_endpoint(switch_eps)
+    if is_supported_combination_button_switch_device_type(device, main_endpoint) then
+      return main_endpoint
+    else
+      return get_first_non_zero_endpoint(button_eps)
```

Comment:
I think this worth logging here because the device has both OnOff and Switch clusters but we're only going to expose the Switch (aka button) in this case. This is likely not what is expected and we should at least log so it's easier to debug.

Reply:
Added a log as suggested!